### PR TITLE
DM-39433: Revert "Move data clients and support code to ts_ess_common"

### PR DESCRIPTION
**Pull request by r-owen**
This reverts commit 18601231f89377b31903a752af47e62e0290702d.

### CODE REVIEW

**Review by wvreeven — APPROVED** [review]
Looks good to me. Please revise the version of the CSC and ESS Common as indicated in my comments.

File: doc/version_history.rst [hunk 1]
```diff
@@ -6,18 +6,17 @@
...
-  This requires ts_ess_common 0.17.
+* Move some documentation to ts_ess_common.
 
 Requires:
 
 * ts_salobj 7
 * ts_idl 3.7
 * IDL file for ESS from ts_xml 16
-* ts_ess_common 0.17
+* ts_ess_common 0.16
```
[the remark this answers is not in the document]
Comment (r-owen) [reply doc/version_history.rst:19]: No, I reverted all visible code changes in ts_ess_common. But your comment prompted me to look at the version history for ts_ess_common and I had forgotten to update it; I have just done that, changing the ts_ess_common version to v0.16.1

File: doc/version_history.rst [hunk 2]
```diff
@@ -6,18 +6,17 @@
 Version History
 ###############
 
-v0.17.0
+v0.16.6
```
[the remark this answers is not in the document]
Comment (r-owen) [reply doc/version_history.rst:9]: The only changes from v0.16.5 are:
* documentation
* self.log.exception("WOUTERRRRR!!!") -> self.log.exception(f"run_data_clients failed: {main_exception!r}")

Unfortunately that is not easy to see in github -- and the same can be said for the other three PRs for this ticket.

[the remark this answers is not in the document]
Comment (wvreeven) [reply doc/version_history.rst:9]: Thanks for the explanation and sorry for having left that exception message in the code.

**Pull request merged**